R/write.R

```diff
@@ -1069,6 +1081,14 @@ write_formula <- function(
   standardize_case_names(...)
 
   assert_class(x, "character")
+
+  # detect array formulas
+  if (any(substr(x, 1, 1) == "{")) {
+    pattern <- "^\\{(.*)\\}$"
+    x <- gsub(pattern, "\\1", x)
+    array <- TRUE
+  }
```

Comment:
I took a confusing and ugly looking function and made it a little more confusing and ugly looking :)

Reply:
regex are always your friends when trying to make a function uglier!